### ISSUE TITLE
react-cosmos componentPaths should handle trailing '/' well

### ISSUE THREAD

**Issue by mocheng** [react-cosmos/react-cosmos#295]
### What's up?

It takes an hour to detect why my "Home" component is displayed as "ome" component and no fixture is found. At last, it turns out to because `componentPaths` requires path not have trailing '/'.

In a word, `componentPaths: ['src/components]` works, while `componentPaths: ['src/components/]` doesn't work.

This looks really... annoying.

### Mkay, tell me more...

BTW, the latest example in this repo doesn't work. It dumps a bunch of error message.
>ERROR in multi main
Module not found: Error: Cannot resolve 'file' or 'directory'

**Comment by mocheng** [comment]
BTW, the `componentPaths` with prefix '/' does work in early 2.0.0-betar release. I made it work a few weeks ago. Then, after a total refresh `npm install`, boooom!

**Comment by skidding** [comment]
Hi @mocheng,

Once `2.0.0` is released `npm install` will no longer break what previously worked. Sorry for going through this.

> It takes an hour to detect why my "Home" component is displayed as "ome" component and no fixture is found. At last, it turns out to because componentPaths requires path not be prefixed with '/'.

[This](https://github.com/react-cosmos/react-cosmos/blob/f94fe30cb4cba9a2cde47c481d7e027f16b1bb37/packages/react-cosmos-voyager/src/index.js#L17) is the culprit. It assumes the path doesn't include a trailing slash so an extra character is removed on top of the base path length.

This is a viable option for fixing this (feel free to come up with something better):

```diff
const extractComponentName = (filePath, rootPath) => {
  let componentName = filePath
-    .slice(rootPath.length + 1)
+    .replace(new RegExp(`^${rootPath}/?(.+)$`), '$1')
    .replace(/\.jsx?$/, '');
```

Let me know if this works for you as well. It would be great if you wanted to make a contribution for this and fix it for everyone. Also let me know if you need any assistance in adding a failing [test case](https://github.com/react-cosmos/react-cosmos/blob/f94fe30cb4cba9a2cde47c481d7e027f16b1bb37/packages/react-cosmos-voyager/src/__tests__/index.js) before applying the fix.

> BTW, the latest example in this repo doesn't work. It dumps a bunch of error message.

Do you mean the `local-storage` example? I just tried it and it worked.

**Comment by mocheng** [comment]
@skidding good to know the info.

I managed to make example work. The problem is that I have a globally installed `webpack` v1.x. After uninstall the global webpack and refresh `npm install`, all examples work fine.

**Comment by skidding** [comment]
Great. I'll leave this open as a `free for all` bug for anybody who wants to put this into master.

**Comment by skidding** [comment]
@mocheng this was fixed in #366
